clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsServerClusterProvider.java

```diff
@@ -533,8 +534,9 @@ public ServerGroup getServerGroup(
 
     try {
       AmazonCredentials credentials = getEcsCredentials(account);
-      // Can't filter by application as there's not enough information in the serverGroupName
-      clusterMap = findClusters(clusterMap, credentials);
+      Moniker moniker = MonikerHelper.applicationNameToMoniker(serverGroupName);
+      log.info("App Name is: " + moniker.getApp());
+      clusterMap = findClusters(clusterMap, credentials, moniker.getApp());
```

Comment:
There used to be a to do here

```
//  TODO - remove the application filter
```

Not sure the reason it was there, but maybe worth restoring that as well?

Reply:
Based on what we're seeing in the current release, I'm a bit hesitant on adding that TODO back